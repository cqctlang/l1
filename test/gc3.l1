@define consume1(g)
{
	@local x;

	x = g();
	if(x != nil)
		printf("guardian returned %a\n", x);
	else
		printf("guardian is empty\n");
	return x;
}

g = mkguardian();
s = "abc";
g(s);
g(s);
g([1,2,3]);
consume1(g);
s = nil;
gc(0,0);

while(consume1(g) != nil)
	;
