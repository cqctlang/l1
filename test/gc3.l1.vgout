guardian is empty
guardian returned [ 1, 2, 3 ]
guardian returned "abc"
guardian returned "abc"
guardian is empty
