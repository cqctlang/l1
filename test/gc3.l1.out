entry: var params, 2 locals, 0 temps
consume1: 1 params, 1 locals, 0 temps
guardian is empty
guardian returned [ 1, 2, 3 ]
guardian returned "abc"
guardian returned "abc"
guardian is empty
